### ISSUE TITLE
Connection refused in the CI

### ISSUE THREAD

**Issue by alecandido** [qiboteam/qibo-cloud-backends#9]
Tests are failing.

https://github.com/qiboteam/qibo-cloud-backends/actions/runs/8084328381/job/22089366544

I just took a quick glance, and I believe you should add the tokens as secrets in the repo, such that you could access them from the workflow.

**Comment by BrunoLiegiBastonLiegi** [comment]
Ah, so you can add some private files in the repo, how do you do that?
Shouls I use my personal IBM token or should we create something like a TII/qibo IBM account?

**Comment by alecandido** [comment]
> Ah, so you can add some private files in the repo, how do you do that?

They are not files, but really tokens. From the *Settings* tab (ask @scarrazza) you can add some key-value pairs, and they are accessible during workflows.

<img width="372" alt="image" src="https://github.com/qiboteam/qibo-cloud-backends/assets/10830876/603b7e33-4202-4d30-b7fc-48dcb17fcebe">

> Shouls I use my personal IBM token or should we create something like a TII/qibo IBM account?

It's not your personal repo, so you should not use your personal token :)

**Comment by scarrazza** [comment]
Thanks, could you please use the new secrets: `IBMQ_TOKEN` and `QIBO_CLIENT_TII_TOKEN`?

**Comment by BrunoLiegiBastonLiegi** [comment]
> Thanks, could you please use the new secrets: `IBMQ_TOKEN` and `QIBO_CLIENT_TII_TOKEN`?

Ok thanks, so are they some kind of shell variables that I can read with `os.environ`?

**Comment by scarrazza** [comment]
Yes, exactly. Load from secrets and assign with environment.

**Comment by BrunoLiegiBastonLiegi** [comment]
Okay, I managed to make the tests run. Unfortunately I had to copy the `rule_poetry.yaml` from the `workflows/` repo as, apparently, environment variables are not supported for reusable workflows, it would have been nice if they mentioned this in the docs....

 Anyway, the `IBMQ` one passes, but the `qibo-client` one gets a refused connection error.

**Comment by scarrazza** [comment]
Does it work outside tests? (I am using the test token that you have)

**Comment by BrunoLiegiBastonLiegi** [comment]
> Does it work outside tests? (I am using the test token that you have)

No I am still getting the connection error:
```
requests.exceptions.ConnectionError: HTTPConnectionPool(host='www.qrccluster.com', port=80): Max retries exceeded with url: /qibo_version/ (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f239d89a2f0>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

By default it's using port 80 though, should I change it?

**Comment by scarrazza** [comment]
No, at the moment we are using 8080.

**Comment by scarrazza** [comment]
Solved in #11.
